auth/authenticators/iam-authenticator.ts

```diff
@@ -14,17 +14,38 @@
  * limitations under the License.
  */
 
-import { OutgoingHttpHeaders } from 'http';
 import { IamTokenManager } from '../token-managers';
 import { validateInput } from '../utils';
-import { BaseOptions, TokenRequestBasedAuthenticator } from './token-request-based-authenticator';
+import { BaseOptions, TokenRequestBasedAuthenticator }
+  from './token-request-based-authenticator';
 
+/** Configuration options for IAM authentication. */
 export interface Options extends BaseOptions {
+  /** The IAM api key */
   apikey: string;
+  /**
+   * The `clientId` and `clientSecret` fields are used to form a "basic"
+   * authorization header for IAM token requests.
+   */
   clientId?: string;
+  /**
+   * The `clientId` and `clientSecret` fields are used to form a "basic"
+   * authorization header for IAM token requests.
+   */
   clientSecret?: string;
 }
 
+/**
+ * The [[IamAuthenticator]] will use the user-supplied `apikey`
+ * values to obtain a bearer token from a token server.  When the bearer token
+ * expires, a new token is obtained from the token server. If specified the
```

Comment:
Two picky grammar changes:

```suggestion
 * expires, a new token is obtained from the token server. If specified, the
```